src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -140,11 +140,6 @@ class profiling_impl
     m_profile_handle = hwctx_handle->open_profile_handle();
   }
 
-  ~profiling_impl()
-  {
-    stop();
```

Comment:
You still require this, but add it under a check if stop is called early , then call stop.